test/CMakeLists.txt

```diff
@@ -67,36 +67,52 @@ if(MIOPEN_TEST_HALF)
     if(MIOPEN_BACKEND_OPENCL)
         set(SKIP_TESTS test_gru test_rnn_vanilla test_lstm test_conv_igemm_dynamic)
     endif()
+    if(MIOPEN_TEST_GFX908)
+        set(SKIP_TESTS test_immed_conv3d test_conv3d test_fusion_aux test_activation test_lrn_test test_ctc test_conv2d_bias test_conv3d_bias test_cba_inference test_cbna_inference test_pooling2d test_na_train test_na_inference test_bn_aux test_conv_igemm_dynamic)
+    endif()
+    if(MIOPEN_TEST_MIOTENSILE)
+        set(SKIP_ALL_EXCEPT_TESTS test_conv2d test_conv3d test_immed_conv2d test_immed_conv3d test_gru test_rnn_vanilla test_lstm test_gru_extra test_rnn_extra test_lstm_extra )
+    endif()
     set(MIOPEN_TEST_FLOAT_ARG --half)
 elseif(MIOPEN_TEST_INT8)
-    set(SKIP_ALL_EXCEPT_TESTS test_tensor_vec test_tensor_cast test_tensor_trans test_tensor_copy test_tensor_set test_tensor_transform test_conv2d)
+    set(SKIP_ALL_EXCEPT_TESTS test_conv2d)
+    if(NOT MIOPEN_TEST_MIOTENSILE)
+        list(APPEND SKIP_ALL_EXCEPT_TESTS test_tensor_vec test_tensor_cast test_tensor_trans test_tensor_copy test_tensor_set test_tensor_transform)
+    endif()
+    list(APPEND SKIP_TESTS test_rnn_extra test_gru_extra test_lstm_extra test_conv_group test_conv_trans test_conv_3d)
     set(MIOPEN_TEST_FLOAT_ARG --int8)
 elseif(MIOPEN_TEST_BFLOAT16)
-    set(SKIP_ALL_EXCEPT_TESTS test_conv2d test_tensor_copy test_tensor_set test_tensor_vec test_immed_conv2d test_check_numerics_test)
-    if(MIOPEN_TEST_GFX908)
-        list(APPEND SKIP_ALL_EXCEPT_TESTS test_conv_extra test_conv_for_implicit_gemm test_miopen_conv test_deepbench_conv)
+    set(SKIP_ALL_EXCEPT_TESTS test_conv2d test_immed_conv2d)
+    if(NOT MIOPEN_TEST_MIOTENSILE)
+        if(MIOPEN_TEST_GFX908)
+            set(SKIP_ALL_EXCEPT_TESTS test_conv_extra test_conv_for_implicit_gemm test_miopen_conv test_deepbench_conv ${SKIP_ALL_EXCEPT_TESTS})
+        endif()
     endif()
+    list(APPEND SKIP_TESTS test_rnn_extra test_gru_extra test_lstm_extra)
     set(MIOPEN_TEST_FLOAT_ARG --bfloat16)
+else()
+    if(MIOPEN_TEST_GFX908)
+        set(SKIP_TESTS test_main test_tensor_scale test_tensor_set test_tensor_transform test_tensor_vec test_w_supertensor test_dropout test_immed_conv3d test_conv3d test_soft_max test_fusion_aux test_activation test_lrn_test test_ctc test_conv2d_bias test_conv3d_bias test_cba_inference test_cbna_inference test_pooling2d test_na_train test_na_inference test_bn_aux test_conv_igemm_dynamic)
+    endif()
+    if(MIOPEN_TEST_MIOTENSILE)
+        set(SKIP_ALL_EXCEPT_TESTS test_conv2d test_conv3d test_immed_conv2d test_immed_conv3d test_gru test_rnn_vanilla test_lstm test_gru_extra test_rnn_extra test_lstm_extra )
+    endif()
```

Comment:
Please explain what you would like to achieve by this change.

Reply:
To skip unnecessary extra tests in miopentensile.

Reply:
Please let me know which tests would you like to skip and will try to re-implement this part of code. Right now it seems overcomplicated.

Reply:
Please let me know which part of the change troubles you and we can discuss how to modify it.

Reply:
I will try to refactor this file as soon as I have time.